feat: add tts provider MiniMax

MindsMaster opened the pull request:
<!-- Put X in the box below to confirm -->

## Checklist:

- [ X] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

MindsMaster committed:
feat: add tts provider MiniMax

Cohee1207 commented on public/scripts/extensions/tts/minimax.js:
```diff
@@ -0,0 +1,990 @@
+        this.settings.customModels.forEach(model => {
+            const modelDiv = $('<div></div>')
+                .css({
+                    'display': 'flex',
```
Consider assigning classes to these elements and place them in a css file of these extension instead of using inline styles

Cohee1207 commented on src/endpoints/secrets.js:
```diff
@@ -54,6 +54,8 @@ export const SECRET_KEYS = {
     SERPER: 'api_key_serper',
     AIMLAPI: 'api_key_aimlapi',
     XAI: 'api_key_xai',
+    MINIMAX: 'api_key_minimax',
```
I don't see these keys being used. Since the requests are not made from the backend.

Cohee1207 commented on public/scripts/extensions/tts/minimax.js:
```diff
@@ -0,0 +1,990 @@
+
+        console.debug('MiniMax TTS Request:', {
+            url: apiUrl,
+            headers: {
```
I'm not sure there's much value in logging these headers since they're static.

Cohee1207 commented on public/scripts/extensions/tts/minimax.js:
```diff
@@ -0,0 +1,990 @@
+                <label for="minimax_tts_api_host">API Host</label>
+                <select id="minimax_tts_api_host" class="text_pole">
+                    <option value="https://api.minimax.chat">Mainland China (api.minimax.chat)</option>
+                    <option value="https://api.minimaxi.chat">Global (api.minimaxi.chat)</option>
```
Can I request the international endpoint to be the default one, pretty please?

Cohee1207 commented on public/scripts/extensions/tts/index.js:
```diff
@@ -138,6 +139,7 @@ const ttsProviders = {
     'TTS WebUI': TtsWebuiProvider,
     VITS: VITSTtsProvider,
     XTTSv2: XTTSTtsProvider,
+    MiniMax: MiniMaxTtsProvider,
```
The providers should be sorted alphabetically.

MindsMaster committed:
Merge branch 'SillyTavern:staging' into staging

MindsMaster committed:
fix: address code review feedback from @Cohee1207

- Moved inline styles to CSS classes in minimax.js
- Removed unused MINIMAX API key from secrets.js
- Removed static headers from debug logs
- Changed default API host to international endpoint
- Sorted TTS providers alphabetically in index.js

MindsMaster commented:
I've implemented all your suggestions. thanks a lot for your feedback!

MindsMaster committed:
feat: improve error handling and add language-specific test sentences based on selected voice

MindsMaster commented:
oh, commit [41dd034](https://github.com/SillyTavern/SillyTavern/pull/4309/commits/41dd0341c77c777d8c616cd17351603d1627a46f) also add a new host "api.minimax.io"

MindsMaster committed:
fix: if no language mapping to MiniMax Format should use "auto"

Cohee1207 commented on public/scripts/extensions/tts/minimax.js:
```diff
@@ -0,0 +1,1145 @@
+    }
+
+    loadCSS() {
+        const cssId = 'minimax-tts-css';
```
Put a [CSS import](https://developer.mozilla.org/en-US/docs/Web/CSS/@import) in [public/scripts/extensions/tts/style.css](https://github.com/SillyTavern/SillyTavern/blob/release/public/scripts/extensions/tts/style.css)

Cohee1207 commented on public/css/minimax-tts.css:
```diff
@@ -0,0 +1,33 @@
+.minimax-custom-item {
```
Move this file under `/public/scripts/extensions/tts/`

MindsMaster replied:
👌

Cohee1207 commented on public/scripts/extensions/tts/minimax.js:
```diff
@@ -0,0 +1,1145 @@
+        });
+
+        try {
+            // Use SillyTavern's CORS proxy to avoid CORS issues
```
It is disabled by default. You need to specify somewhere in the extension settings that it is required to be enabled with config.

Cohee1207 replied:
Alternative (that I prefer more): move all API communication to Node backend

MindsMaster replied:
I’ll fix it up later, I originally had a separate request script running on the backend then deleted it🤦‍♂️

MindsMaster committed:
fix: move this minimax-tts.css under /public/scripts/extensions/tts/

MindsMaster committed:
refactor: move minimax api communication to Node backend

MindsMaster commented:
I've completed all the changes based on your review. Thank you very much for your guidance

MindsMaster committed:
chore: unify import style for minimax router

Cohee1207 commented on src/server-startup.js:
```diff
@@ -45,6 +45,7 @@ import { router as koboldRouter } from './endpoints/backends/kobold.js';
 import { router as textCompletionsRouter } from './endpoints/backends/text-completions.js';
 import { router as speechRouter } from './endpoints/speech.js';
 import { router as azureRouter } from './endpoints/azure.js';
+import minimaxRouter from './endpoints/minimax.js';
```
Default export for router sores my eye :(

MindsMaster replied:
its already fix 
commit 55ef2bd90456cc4c979a0f6378a09fdbab1243cf
sry bro

Cohee1207 committed:
Move css into subfolder

Cohee1207 committed:
More secure secret key handling

Cohee1207 committed:
Better UI layout

Cohee1207 committed:
Finish secrets update

Cohee1207 committed:
Replace with a normal import

Cohee1207 committed:
This block shouldn't be here

Cohee1207 reviewed (approved):
This requires documentation. It's unclear how to add voices

MindsMaster commented:
You can obtain the GroupId and APIKey from the MiniMax API console and fill them in.
Each voice provided by the official service has its own VoiceId, which can be found on the voice selection page.
<img width="670" height="289" alt="a0c885c5-812a-4eb4-b12d-9012e3e83fb6" src="https://github.com/user-attachments/assets/bbf9fe70-ada8-4258-a1d7-7b549c709f2d" />

Cohee1207 commented:
Can you create a page and submit it here?
https://github.com/SillyTavern/SillyTavern-Docs

MindsMaster commented:
sure👌

Cohee1207 commented:
I'm unable to test it since it requires $25 payment and I'm not paying that for TTS, so it's good to go as-is.
